### PR TITLE
Clean up VcsSim init()

### PR DESCRIPTION
**Pull request by timsifive**
Use a unique log file, so you can run multiple instances at once.
Add time out to waiting for the simulator to be ready.

### CODE REVIEW

File: debug/testlib.py [hunk 1]
```diff
@@ -142,9 +142,10 @@ def wait(self, *args, **kwargs):
         return self.process.wait(*args, **kwargs)
 
 class VcsSim(object):
-    logname = "simv.log"
+    logfile = tempfile.NamedTemporaryFile(prefix='simv', suffix='.log')
+    logname = logfile.name
 
-    def __init__(self, sim_cmd=None, debug=False):
+    def __init__(self, sim_cmd=None, debug=False, timeout=300):
```
Comment (mwachs5) [inline debug/testlib.py]: is there a path to override this timeout? Let's not hard-code any timeouts.

Comment (mwachs5) [reply debug/testlib.py]: For example can target class have a "init_timeout" variable which we pass in here but can be overridden in the target subclass?

Comment (timsifive) [reply debug/testlib.py]: Yes. The function is called from the .py file containing a target configuration, so each target can specify its own timeout.

**Comment by timsifive** [comment]
The timeout could be configured eg. at https://github.com/riscv/riscv-tests/blob/master/debug/targets/SiFive/Freedom/E300Sim.py#L17

**Pull request merged**